tools/crm_shadow.c

```diff
@@ -289,40 +289,56 @@ write_shadow_file(xmlNode *xml, const char *filename, bool reset,
     return pcmk_rc_ok;
 }
 
+/*!
+ * \internal
+ * \brief Set up environment variables for a shadow instance
+ *
+ * \param[in]  do_switch  If true, switch to an existing instance (logging only)
+ * \param[out] error      Where to store error
+ */
 static void
-shadow_setup(char *name, gboolean do_switch)
+shadow_setup(bool do_switch, GError **error)
 {
+    const char *active = getenv("CIB_shadow");
     const char *prompt = getenv("PS1");
     const char *shell = getenv("SHELL");
-    char *new_prompt = get_shadow_prompt(name);
-
-    printf("Setting up shadow instance\n");
```

Comment:
It looks good to me. All the `crm_shadow` caller code is in here: https://github.com/ClusterLabs/crmsh/blob/master/crmsh/ui_cib.py

It would be in `do_new()` or `do_reset()` (there is no direct support for `--switch`... the closest thing is `do_use()`). Everything except `do_diff()` seems to discard stdout though.

Reply:
Great